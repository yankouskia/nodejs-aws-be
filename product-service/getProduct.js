import { loadProduct } from './db';

export const getProduct = async event => {
  try {
    console.log('Event Path Parameters', event.pathParameters);

    const { id } = event.pathParameters;
    const product = await loadProduct(id);

    if (!product) {
      return {
        statusCode: 200,
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Access-Control-Allow-Credentials': true,
        },
        body: JSON.stringify({ product, message: 'Product not found by passed ID' }),
      };
    }

    console.log('Found result', product);

    return {
      statusCode: 200,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
      },
      body: JSON.stringify({ product }),
    };
  } catch (e) {
    console.log('Database or Server Error', e);

    return {
      statusCode: 500,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
      },
      body: JSON.stringify({ message: 'Error while reading data' }),
    };
  }
};
