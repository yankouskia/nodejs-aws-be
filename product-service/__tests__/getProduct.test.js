import { getProduct } from '../getProduct';

jest.mock('../db.js', () => ({
  loadProduct: () => ({
    id: 'product',
  })
}))

test('should return product', async () => {
  const product = await getProduct({ pathParameters: { id: '1' }});
  expect(product.statusCode).toEqual(200);
  expect(product.body).toMatchSnapshot();
});

test('should return error if no path parameter', async () => {
  const product = await getProduct();
  expect(product.statusCode).toEqual(500);
  expect(product.body).toMatchSnapshot();
});
