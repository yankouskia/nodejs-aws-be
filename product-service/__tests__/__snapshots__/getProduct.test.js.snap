// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return error if no path parameter 1`] = `"{\\"message\\":\\"Error while reading data\\"}"`;

exports[`should return product 1`] = `"{\\"product\\":{\\"id\\":\\"product\\"}}"`;
